src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/BuildIntegratedProjectAction.cs

```diff
@@ -104,6 +117,63 @@ public class BuildIntegratedProjectAction : NuGetProjectAction
             Sources = sources;
             OriginalActions = originalActions;
             InstallationContext = installationContext;
+            ActionAndContextList = originalActions.Select(e => (e, installationContext)).ToList();
+        }
+
+        internal BuildIntegratedProjectAction(NuGetProject project,
+            PackageIdentity packageIdentity,
+            NuGetProjectActionType nuGetProjectActionType,
+            LockFile originalLockFile,
+            RestoreResultPair restoreResultPair,
+            IReadOnlyList<SourceRepository> sources,
+            IReadOnlyList<(NuGetProjectAction, BuildIntegratedInstallationContext)> originalActionsAndInstallationContexts,
+            VersionRange versionRange)
+            : base(packageIdentity, nuGetProjectActionType, project, sourceRepository: null, versionRange)
+        {
+            if (project == null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+
+            if (packageIdentity == null)
+            {
+                throw new ArgumentNullException(nameof(packageIdentity));
+            }
+
+            if (originalLockFile == null)
+            {
+                throw new ArgumentNullException(nameof(originalLockFile));
+            }
+
+            if (restoreResultPair == null)
+            {
+                throw new ArgumentNullException(nameof(restoreResultPair));
+            }
+
+            if (sources == null)
+            {
+                throw new ArgumentNullException(nameof(sources));
+            }
+
+            if (originalActionsAndInstallationContexts == null)
+            {
+                throw new ArgumentNullException(nameof(originalActionsAndInstallationContexts));
+            }
+
+            if (originalActionsAndInstallationContexts.Count < 1)
+            {
+                throw new ArgumentException("Must contain at least 1 element.", nameof(originalActionsAndInstallationContexts));
```

Comment:
I think we should localize this exception message.

Reply:
This is an internal constructor. 
If this happens, it's our mistake.